taichi/transforms/simplify.cpp

```diff
@@ -691,7 +688,32 @@ class BasicBlockSimplify : public IRVisitor {
         }
       }
     }
-    set_done(stmt);
+
+    // Lower into a series of adds and muls.
+    // Need typecheck afterwards.
+    if (stmt->inputs.empty()) {
```

Comment:
Actually, no need to specialize when you have empty inputs. Empty inputs should simply be treated as `inputs` with zero lengths. Then you can merge two clauses of this `if` stmt. The extra `0 + i` can be optimized by your `alg_simp` pass.